bot/api.py

```diff
@@ -58,3 +62,76 @@ def maybe_raise_for_status(self, response: aiohttp.ClientResponse, should_raise:
 
             self.maybe_raise_for_status(resp, raise_for_status)
             return await resp.json()
+
+
+def loop_is_running() -> bool:
+    # asyncio does not have a way to say "call this when the event
+    # loop is running", see e.g. `callWhenRunning` from twisted.
+
+    try:
+        asyncio.get_running_loop()
+    except RuntimeError:
+        return False
+    return True
+
+
+class APILoggingHandler(logging.StreamHandler):
+    def __init__(self, client: APIClient):
+        logging.StreamHandler.__init__(self)
```

Comment:
Why not `super()`? Don't trust its black magic?

Reply:
Nope, nope, nope